src/lib/y2network/clients/routing_auto.rb

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "installation/auto_client"
+require "y2network/config"
+require "y2network/routing"
+require "y2network/autoyast_routing"
+require "y2network/routing_helpers"
+
+Yast.import "NetworkInterfaces"
+Yast.import "Lan"
+
+module Y2Network
+  module Clients
+    # This class is reponsible of the autoinstallation routing configuration.
+    class RoutingAuto < ::Installation::AutoClient
+      include RoutingHelpers
+      include Yast::I18n
+      include Yast::Logger
+
+      class << self
+        # @return [Boolean] whether the AutoYaST configuration has been
+        # modified or not
+        attr_accessor :changed
+      end
+
+      # Constructor
+      def initialize
+        textdomain "network"
+      end
+
+      # Import routing configuration
+      #
+      # @param profile [Hash] routing profile section to be imported
+      # @return [Boolean]
+      def import(profile = {})
+        return unless config
+
+        config.routing = ay_routing.import(profile)
+        modified!
+        config.routing
```

Comment:
it should return boolean according to doc, not?

Reply:
Yes, I just returned routing for testing, but will modify it. The ay_routing will return A [Routing] object